mycpp/gc_dict.h

```diff
@@ -231,25 +202,25 @@ void Dict<K, V>::reserve(int n) {
     new_k = NewSlab<K>(capacity_);
```

Comment:
line 196 can have its comment removed

I think we want to make `index_` bigger than `keys_` and `values_`, maybe multiply by 4/3 or 3/2 ?  

So the load factor is  3/4 or 2/3 (at most 3 out of 4 entries will not be kEmptyEntry or kDeletedEntry)


In theory this should affect performance -- the probing should be shorter

Reply:
Went with 3/2 as a multiplier for now. This was basically a random choice, though. We should profile at some point and tweak accordingly